assignment4.Rmd

```diff
@@ -105,12 +114,33 @@ sweep(loadings, 2, colSums(loadings), "/") #sweep() computes each row as a propo
 
 #Now examine your components and try to come up with substantive descriptions of what some might represent?
 
+# these components convert the originial data into a new space,  maybe some can present study motivation, and some can represent basic knowledge level.
+
 #You can generate a biplot to help you, though these can be a bit confusing. They plot the transformed data by the first two components. Therefore, the axes represent the direction of maximum variance. Then mapped onto this point cloud are the original directions of the variables, depicted as red arrows. It is supposed to provide a visualization of which variables "go together". Variables that possibly represent the same underlying construct point in the same direction.  
 
 biplot(pca)
 
 #Calculate values for each student that represent these your composite variables and then create a new correlogram showing their relationship to mean_correct.
-
+D5<-pca$x[,1:5]
+D6 <- cbind(D5, as.data.frame(D1$mean_correct))
+COR3<-cor(D6)
+corrplot(COR3, order="AOE", method="circle", tl.pos="lt", type="upper",        
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
 ```
 # Part III  
 ## Also in this repository is a data set and codebook from Rod Martin, Patricia Puhlik-Doris, Gwen Larsen, Jeanette Gray, Kelly Weir at the University of Western Ontario about people's sense of humor. Can you perform a PCA on this data?
+
+```{r}
+humor<-read.csv("humor_data.csv")
+dim(humor)
+Dh<-humor[,1:36]
+Dh<-scale(Dh,center=T)
+pcah<-prcomp(Dh,scale=TRUE)
+summary(pcah)
+plot(pcah,type='lines')
+biplot(pcah)
```

Comment:
Great coding 👍 Can you interpret your results? Did you see the same patterns as the original research?